src/schemathesis/extra/pytest_plugin.py

```diff
@@ -106,13 +117,35 @@ def collect(self) -> List[Function]:  # type: ignore
 
 
 class SchemathesisFunction(Function):  # pylint: disable=too-many-ancestors
+    def __init__(self, *args: Any, test_func: Callable, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.test_function = test_func
+
     def _getobj(self) -> partial:
         """Tests defined as methods require `self` as the first argument.
 
         This method is called only for this case.
         """
         return partial(self.obj, self.parent.obj)
 
+    def get_more_tests(self) -> List["SchemathesisFunction"]:
+        return [
+            create(
+                self.__class__,
+                name=f"{self.originalname}[{e.method}:{e.full_path}]",
```

Comment:
Improve the displayed name, we need to indicate that this test comes from the original one.